src/sql/workbench/services/commandLine/common/commandLineService.ts

```diff
@@ -136,11 +137,37 @@ export class CommandLineService implements ICommandLineProcessing {
 		profile.serverName = args.server;
 		profile.databaseName = args.database ? args.database : '';
 		profile.userName = args.user ? args.user : '';
-		profile.authenticationType = args.integrated ? 'Integrated' : 'SqlLogin';
+		profile.authenticationType = args.integrated ? Constants.integrated : args.aad ? Constants.azureMFA : (profile.userName.length > 0) ? Constants.sqlLogin : Constants.integrated;
 		profile.connectionName = '';
 		profile.setOptionValue('applicationName', Constants.applicationName);
 		profile.setOptionValue('databaseDisplayName', profile.databaseName);
 		profile.setOptionValue('groupId', profile.groupId);
-		return profile;
+		return this._connectionManagementService ? this.tryMatchSavedProfile(profile) : profile;
+	}
+
+	private tryMatchSavedProfile(profile: ConnectionProfile)
+	{
+		let match: ConnectionProfile = undefined;
+		// If we can find a saved mssql provider connection that matches the args, use it
+		let groups = this._connectionManagementService.getConnectionGroups([Constants.mssqlProviderName]);
+		if (groups && groups.length > 0)
+		{
+			let rootGroup = groups[0];
+			let connections = rootGroup.connections;
```

Comment:
Why do you just look in the root group? You don't want to try and match connections in custom groups?

Reply:
fixed